.github/workflows/linux-publish.yml

```diff
@@ -0,0 +1,172 @@
+    env:
+      BUILD_CONFIG: "Release"
+    steps:
+      - name: Checkout Workspace
+        uses: actions/checkout@v4
+
+      - uses: actions/setup-node@v4
+        with:
+          node-version: 18
+          registry-url: https://npm.pkg.github.com/
+
+      - name: build app-ziti-console
+        shell: bash
+        run: ./build.sh /zac/
+
+      - name: upload app-ziti-console
+        uses: actions/upload-artifact@v4
+        with:
+          name: app-ziti-console
+          path: ./dist/app-ziti-console/
+
+  publish-linux-packages:
+    needs:
+      - parse-version
+      - build-and-upload-app-ziti-console
+    name: ${{ matrix.package_name }} ${{ matrix.arch.rpm }} ${{ matrix.nfpm_packager }}
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        package_name:
+          - openziti-console
+        arch:
+          - goreleaser: amd64
+            deb: amd64
+            rpm: x86_64
+          - goreleaser: arm64
+            deb: arm64
+            rpm: aarch64
+          - goreleaser: armv7
+            deb: armv7
+            rpm: armv7
+        nfpm_packager:
+          - rpm
+          - deb
+          # - archlinux #  (pacman)
+    env:
+      ZAC_VERSION: ${{ needs.parse-version.outputs.version }}
+      ZITI_MAINTAINER: "OpenZiti Maintainers <developers@openziti.org>"
+      ZITI_HOMEPAGE: "https://openziti.io"
+      ZITI_VENDOR: "NetFoundry"
+      GOARCH: ${{ matrix.arch.goreleaser }}
+      ZITI_DEB_TEST_REPO: ${{ vars.ZITI_DEB_TEST_REPO || 'zitipax-openziti-deb-test' }}
+      ZITI_RPM_TEST_REPO: ${{ vars.ZITI_RPM_TEST_REPO || 'zitipax-openziti-rpm-test' }}
+      ZITI_DEB_PROD_REPO: ${{ vars.ZITI_DEB_PROD_REPO || 'zitipax-openziti-deb-stable' }}
+      ZITI_RPM_PROD_REPO: ${{ vars.ZITI_RPM_PROD_REPO || 'zitipax-openziti-rpm-stable' }}
+    steps:
+      - name: Checkout Workspace
+        uses: actions/checkout@v4
+
+      - name: Download app-ziti-console
+        uses: actions/download-artifact@v4
+        with:
+          name: app-ziti-console
+          path: ./dist/app-ziti-console
+
+      - name: Build Package
+        id: nfpm
+        uses: burningalchemist/action-gh-nfpm@v1
+        with:
+          nfpm_version: "2.38.0"
+          packager: ${{ matrix.nfpm_packager }}
+          config: ./linux-packages/openziti-console/nfpm-${{ matrix.package_name }}.yaml
+          target: ./dist/
+
+      - name: upload package artifact to build summary page
+        uses: actions/upload-artifact@v4
+        with:
+          name: ${{ matrix.package_name }}-${{ matrix.arch.goreleaser }}-${{ matrix.nfpm_packager }}
+          path: ./dist/${{ matrix.package_name }}*.${{ matrix.nfpm_packager }}
+          if-no-files-found: error
+
+      - name: Configure jFrog CLI
+        uses: jfrog/setup-jfrog-cli@v4
+        env:
+          JF_ENV_1: ${{ secrets.ZITI_ARTIFACTORY_CLI_CONFIG_PACKAGE_UPLOAD }}
+        with:
+          version: 2.63.0
+
+      - name: Upload RPM to Artifactory testing repo
```

Comment:
is this and the other testing repo upload really necessary?

Reply:
In this case, uploading to the testing repo means that machines / testers that have bravely subscribed to the testing repo will have access to this version. When this workflow runs on a release it copies the artifact to the prod repo for everyone else. This simplifies setup for testers because they don't need to configure both test and prod repos to have access to all versions. It's a pattern across all our repos.

Reply:
Right. and my point is it's not necessary. I'd like to see demand for this functionality before just doing it

Reply:
It's a pattern across all our repos that releases are available in both test and stable repos. If it wasn't uploaded to the test repo, then testers wouldn't be able to install it. I'm one of the testers, so it would make my life harder.

Reply:
i don't want the project to have the additional burden of supporting this deployment mode. if you want the latest, you should get it from the repository. we don't need to be deploying every push to main

Reply:
It runs on push to main to check that the package can still be built and stops short of uploading it. Only releases upload. Your question here seemed to be "why upload releases to test repo first?"